### PR TITLE
namspace.js no longer needed in Drupal

### PR DESCRIPTION
**Pull request by LightGuard**
See https://issues.jboss.org/browse/DEVELOPER-3144 for more information.

There is a new file drupal-namespace.js which is being used for Drupal,
in combination with some javascript coming from Drupal based on the
environment.

I have removed the scripts from the twig files where I had them in place
to get things working. namespace.js is also no longer being sent with
pages being rendered by awestruct and input into Drupal. This removed
the need for the as_is_html format so I removed that as well.

What we're currently using for client side html templates are now stored
in twig files (but they're not being rendered via twig). If you need to
make changes you will have to make changes in two places, sorry.

I've also added some variables into the yml files in the different
environments that were being taken care of via _config/site.yml in
awestruct.

Lastly, I've updated Drupal to 8.1.2 and also updated some of the
contrib modules to their latest releases.

### CODE REVIEW

**Comment by LightGuard** [comment]
retest this please

**Comment by LightGuard** [comment]
@robpblake please review and verify JS in Drupal is working correctly. @Dantheman720 feel free to give it a try too.

**Comment by robpblake** [comment]
@LightGuard As per our discussions in chat. I'm not seeing any content previously loaded by JS now loading. Looks like all of the URLs are null in the drupalSettings.rhd Javascript object.

cheers,

Rob

**Comment by LightGuard** [comment]
retest this please

**Comment by Dantheman720** [comment]
@LightGuard @robpblake - The JS seems to be working now. Pending successful build, it should be good to go.

**Comment by LightGuard** [comment]
For those of you using this, you may need to do a `composer update` but I hope not.

**Comment by LightGuard** [comment]
retest this please

**Comment by LightGuard** [comment]
@adelasofia this should be good now, @Dantheman720 has verified on his box that this is working properly.

**Comment by robpblake** [comment]
@LightGuard A couple of comments:

JS now seems to be working across the site as per other comments. A couple of things I noticed:

1.  The upgrade to 8.2.4 of the simple_sitemap module breaks the export functionality. It looks like the implementation of the module has changed significantly from 8.2.3 and none of the content is pushed into the sitemap.xml by default any more (I suspect our custom configuration needs updating to match what 8.2.4 expects). I'd suggest reverting back to 8.2.3 for your PR and we can fix that in another PR (for example I can do in the work to make exports quicker)

2. http://docker:32769/web-and-api-development/overview/ This page is now completely broken for me after '--run-the-stack'. I'm not sure if this is related to the changes in this PR, but perhaps work a quick check to make sure.

**Comment by LightGuard** [comment]
@robpblake the issues you've discovered should now be fixed.

**Comment by LightGuard** [comment]
retest this please

File: _docker/drupal/web/themes/custom/rhd/templates/solution/page--rhd-solution-overview.html.twig [hunk 1]
```diff
@@ -80,13 +82,14 @@ view_mode ? 'node--view-mode-' ~ view_mode|clean_class,
                     <meta itemprop="position" content="1">
                 </li>
                 <li itemprop="itemListElement" itemscope="" itemtype="http://schema.org/ListItem">
-                    <a href="http://docker:32769/web-and-api-development/" itemprop="item">{{ content.field_solution_name }}</a>
+                    <a href="http://docker:32769/{{ solution_name_hyphen }}/"
```
Comment (robpblake) [inline _docker/drupal/web/themes/custom/rhd/templates/solution/page--rhd-solution-overview.html.twig]: This shouldn't be hardcoded to 'docker:32769'

File: _docker/lib/rhd_environment.rb [hunk 2]
```diff
@@ -72,7 +72,7 @@ def get_file(file_name)
   # that needs this is the Drupal pull-request environment
   #
   def template_resources
-    if environment_name == 'drupal-pull-request'
+    if is_drupal_environment?
```
Comment (robpblake) [inline _docker/lib/rhd_environment.rb]: Can you explain the reason we need to template resources in any other environment other than the drupal-pull-request environment? As far as I can think of, the drupal-pull-request environment is the only one that would require some form of templating as we don't know the drupal or searchisko ports until run-time. 

Is there another scenario where we need to template?

**Comment by robpblake** [comment]
retest this please

**Comment by robpblake** [comment]
@LightGuard Pages are now loaded OK, but I'm seeing things like "Header", "Left Section" etc on the layouts. See attached screenshots:

screen shot 2016-06-14 at 11 06 41

screen shot 2016-06-14 at 11 04 58

screen shot 2016-06-14 at 11 07 23

File: _docker/drupal/web/themes/custom/rhd/templates/solution/page--rhd-solution-overview.html.twig (hunk 1 above)
Comment (LightGuard) [inline _docker/drupal/web/themes/custom/rhd/templates/solution/page--rhd-solution-overview.html.twig]: Good catch

**Comment by LightGuard** [comment]
retest this please

**Comment by LightGuard** [comment]
retest this please

**Comment by LightGuard** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by robpblake** [comment]
@LightGuard It would now appear that the header banner and left-hand menu are missing on the overview pages. As an example:

http://developers-pr.stage.redhat.com/pr/1161/build/1417/containers/overview/

screen shot 2016-06-15 at 15 27 32

Same page after running `--run-the-stack` in the `drupal-dev` environment on my machine:

screen shot 2016-06-15 at 15 28 54

**Comment by LightGuard** [comment]
@robpblake updated. All the overview pages should be correct now.

**Comment by robpblake** [comment]
retest this please

**Comment by robpblake** [comment]
@LightGuard This is partly fixed but some of the pages still have the layout problem. For example the layout is still not correct on:

- http://docker:32769/containers/adoption/
- http://docker:32769/mobile/adoption/

I'm also seeing a different set of links on the containers overview page locally than I do on the PR site:

http://developers-pr.stage.redhat.com/pr/1161/build/1422/containers/overview/:

screen shot 2016-06-15 at 18 15 56

http://docker:32769/containers/overview/
screen shot 2016-06-15 at 18 16 39

The last two links on the screenshot above 404.

### To test and replicate locally:

I'm using the following steps to find these issues:

1. Click on the "Topics" main heading
2. Select any of the sub-menu items e.g. Containers
3. Click on the "Get Started" button on the right-hand side
4. Click through the left-hand menu items e.g. Overview, Adoption, Learn, Buzz

You'll see that some pages have the correct layout and others not.

**Comment by LightGuard** [comment]
The changes should only affect the overview pages. Adoption pages should be the same.

**Comment by Dantheman720** [comment]
+1 Looks good to me. JS on the resources, buzz, latest and connectors pages work as intended. Everything works similar to production (aside from the persistence issue that causes the strange headers to appear at the top).

**Pull request merged**